src/app/components/auth/signup/RegisterForm.tsx

```diff
@@ -0,0 +1,370 @@
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({ name, email, password }),
+      });
+
+      if (response.status === 200) {
+        try {
+          const response = await fetch('/api/auth/', {
+            method: 'POST',
+            headers: {
+              Accept: 'application/json',
+              'Content-Type': 'application/json',
+            },
+            body: JSON.stringify(data),
+          });
+
+          if (response.status === 200) {
+            router.replace('/');
+          }
+        } catch (e) {
+          router.push('/auth/sign-in');
+        }
+      } else {
+        const data = (await response.json()) as ErrorResponse;
+        response.status === 401
+          ? setSignUpError(data.errorCode)
+          : setSignUpError(SignUpErrorCodes.UnknownError);
+      }
+    } catch (e) {
+      if (e instanceof Error) {
+        setSignUpError(SignUpErrorCodes.UnknownError);
+      }
+    }
+  }
+
+  return (
+    <form
+      className="space-y-8"
+      onSubmit={handleSubmit(handleSubmitEvent)}
+      noValidate
+    >
+      <p className="text-sm text-center font-light text-gray-500 dark:text-gray-400">
+        {alreadyHaveAccount}{' '}
+        <Link
+          href={'/auth/sign-in'}
+          className="font-medium text-gray-700 dark:text-gray-400 focus:underline hover:underline"
+          data-testid="signin"
+        >
+          {signinLink}
+        </Link>
+      </p>
+      <div className="relative">
+        <label
+          htmlFor="name"
+          className="block mb-2 text-sm font-medium text-gray-900 dark:text-white"
+        >
+          {nameLabel}
+        </label>
+        <input
+          type="text"
+          name={nameName}
+          ref={nameRef}
+          onChange={onNameChange}
+          onBlur={onNameBlur}
+          className="mb-1 block w-full px-3 py-2 border rounded-lg dark:border-gray-700 dark:bg-gray-900 dark:text-gray-100 focus:dark:border-violet-400"
+          placeholder={namePlaceholder}
+          aria-label="name"
+          data-testid="name"
+        />
+        {errors.name?.message && (
+          <ErrorMessage
+            message={
+              locale[currentLang][errors.name?.message as formValidationErrors]
+            }
+          />
+        )}
+      </div>
+      <div className="relative">
+        <label
+          htmlFor="email"
+          className="block mb-2 text-sm font-medium text-gray-900 dark:text-white"
+        >
+          {emailLabel}
+        </label>
+        <input
+          type="email"
+          name={emailName}
+          ref={emailRef}
+          onChange={onEmailChange}
+          onBlur={onEmailBlur}
+          className="mb-1 block w-full px-3 py-2 border rounded-lg dark:border-gray-700 dark:bg-gray-900 dark:text-gray-100 focus:dark:border-violet-400"
+          placeholder="name@company.com"
+          aria-label="email"
+          data-testid="email"
+        />
```

Comment:
You can create RHFInput component which uses `useFormContext` hook from RHF.  From that hook you'll be able to provide all needed props for input. 
To be able to use that hook, you need to use form-provider which is said about in above comment

Reply:
@iamnkt
Please take this task.